src/components/core/geoChart/tests/GeoChartLegendRenderer.spec.tsx

```diff
@@ -15,34 +20,63 @@ function createComponent(customProps: IGeoChartLegendRendererProps) {
+            ? {
+                  index: 0,
+                  name: "size",
+              }
+            : undefined,
+        color: hasColorLegend
+            ? {
+                  index: hasSizeLegend ? 1 : 0,
+                  name: "color",
+              }
+            : undefined,
+        location: {
+            index: 0,
+            name: "location",
+        },
+    };
+    const executionResult = getExecutionResult(true, false, false, hasSizeLegend, hasColorLegend);
+    const { sizeData, colorData } = calculateLegendData(executionResult, geoData);
+    const sizeFormat = "#,##0";
+    const colorFormat = "#,##0";
+    return {
+        sizeData,
+        sizeFormat,
+        colorData,
+        colorFormat,
+        isColorLegendVisible: hasColorLegend,
+        config: getGeoConfig({ isWithLocation: true, isWithSize: true }),
+        locale: "en-US",
+    };
+}
+
 describe("GeoChartLegendRenderer", () => {
-    it("should render component with PushpinSizeLegend", () => {
-        const execution: Execution.IExecutionResponses = {
-            executionResponse: getExecutionResponse(true, false, false, true),
-            executionResult: getExecutionResult(true, false, false, true),
-        };
-        const props: IGeoChartLegendRendererProps = {
-            execution,
-            config: getGeoConfig({ isWithLocation: true, isWithSize: true }),
-            locale: "en-US",
-        };
-        const wrapper = createComponent(props);
-        const legend = wrapper.find(PushpinSizeLegend);
-        expect(wrapper.find(".s-geo-legend")).toHaveLength(1);
-        expect(legend.length).toEqual(1);
+    it("should not render component", () => {
+        const wrapper = createComponent(getLegendProps({}));
+        expect(wrapper.find(".s-geo-legend")).toHaveLength(0);
     });
 
-    it("should not render component ", () => {
-        const execution: Execution.IExecutionResponses = {
-            executionResponse: getExecutionResponse(true),
-            executionResult: getExecutionResult(true),
-        };
-        const props: IGeoChartLegendRendererProps = {
-            execution,
-            config: getGeoConfig({ isWithLocation: true }),
-            locale: "en-US",
-        };
-        const wrapper = createComponent(props);
-        expect(wrapper.find(".s-geo-legend")).toHaveLength(0);
+    it("should render component with PushpinSizeLegend", async () => {
+        const wrapper = createComponent(getLegendProps({ hasSizeLegend: true }));
+        expect(await wrapper.find(".s-geo-legend")).toHaveLength(1);
+        expect(await wrapper.find(".s-pushpin-size-legend")).toHaveLength(1);
+    });
+
+    it("should render component with color legend", async () => {
+        const wrapper = createComponent(getLegendProps({ hasColorLegend: true }));
+        expect(await wrapper.find(".s-geo-legend")).toHaveLength(1);
+        expect(await wrapper.find(".heatmap-legend")).toHaveLength(1);
```

Comment:
why do we check for `heatmap-legend` here?
Please add a general classname for both heatmap and geo chart, with prefix `s-`
:)

Reply:
it might be a cause of failed graphene and testcafe tests on AD, KD. I would keep the class name the same. And then, we can improve when we have geoChart on AD and KD.